backend/src/ast/converters/scratch/scratch-sound-block-converter.ts

```diff
@@ -0,0 +1,103 @@
+import { CodeNode } from "src/ast/types/general-ast/ast-nodes";
+import { match, P } from "ts-pattern";
+import { NonHatBlock, TreeNode } from "./types";
+import {
+  isChangeEffectByBlock,
+  ChangeEffectByBlock,
+  ChangeVolumeByBlock,
+  ClearEffectsBlock,
+  isChangeVolumeByBlock,
+  isClearEffectsBlock,
+  isPlayBlock,
+  isPlayUntilDoneBlock,
+  isSetEffectToBlock,
+  isSetVolumeToBlock,
+  isStopAllSoundsBlock,
+  isVolumeBlock,
+  PlayBlock,
+  PlayUntilDoneBlock,
+  SetEffectToBlock,
+  SetVolumeToBlock,
+  SoundCodeBlock,
+  StopAllSoundsBlock,
+  VolumeBlock,
+  SoundExpressionBlock,
+  isSoundsMenuBlock,
+  SoundsMenuBlock,
+} from "src/ast/types/input/scratch/blocks/sound";
+import {
+  createFunctionCallBlock,
+  createLiteralNode,
+  createVariableExpressionBlock,
+} from "./helpers";
+import { ExpressionNode } from "src/ast/types/general-ast/ast-nodes/code-node/expression-node";
+
+type SoundCodeTreeNode = SoundCodeBlock & TreeNode;
+type SoundExpressionTreeNode = SoundExpressionBlock & TreeNode;
+
+export const isSoundCodeBlock = (
+  block: NonHatBlock,
+): block is SoundCodeBlock & NonHatBlock =>
+  isSoundsMenuBlock(block) ||
```

Comment:
`isSoundsMenuBlock` shows up both here and below. Is this by design ?

Reply:
no that was not intentional, good catch!